src/ceph_mds.cc

```diff
@@ -244,6 +244,7 @@ int main(int argc, const char **argv)
   // detection, etc.).  don't bother if it was a suicide.
   if (mds->is_clean_shutdown()) {
     delete mds;
+    msgr->wait();
```

Comment:
For the case you're dealing with, you would want to add the msgr->wait() up at line 216, before "goto shutdown".  Adding it here doesn't work (is_clean_shutdown is something completely different)